src/status_im/contexts/wallet/wallet_connect/processing_events.cljs

```diff
@@ -126,38 +126,37 @@
 (rf/reg-event-fx
  :wallet-connect/process-sign-typed
  (fn [{:keys [db]}]
-   (let [[address raw-data] (wallet-connect-core/get-db-current-request-params db)
-         parsed-raw-data    (transforms/js-parse raw-data)
-         session-chain-id   (-> (wallet-connect-core/get-db-current-request-event db)
-                                (get-in [:params :chainId])
-                                wallet-connect-core/eip155->chain-id)
-         data-chain-id      (-> parsed-raw-data
-                                transforms/js->clj
-                                signing/typed-data-chain-id)
-         parsed-data        (try (-> parsed-raw-data
-                                     (transforms/js-dissoc :types :primaryType)
-                                     (transforms/js-stringify 2))
-                                 (catch js/Error _ nil))]
-     (cond
-       (nil? parsed-data)
+   (try
```

Comment:
Where can the exception be thrown? If we could add the try/catch surrounding only what can fail it would make clear where the problem can be.

Reply:
I was wondering how to do this in a cleaner way. The exception can only be thrown when parsing the JSON string here:
```clj
   (let [[address raw-data] (wallet-connect-core/get-db-current-request-params db)
         parsed-raw-data    (transforms/js-parse raw-data) ;; HERE
```
(...), but when catching we should short-circuit and return the dispatch fx. Do you have a suggestion on how to isolate it cleanly?

Reply:
If it's the first thing to happen, you could extract to a separate function, catch and return nil and use `when-let`.

Reply:
but then we lose on the actual error, which would make it more difficult to understand the problem, no?

Reply:
Sorry I didn't look too deep.

We could return a tuple `[response, error]` like other languages do. Then that would give you room to reuse the values and also check if error exists.

```clojure
(let [[[address raw-data] err] (dangerous!)]
  (if err
    (error-effects)
    (success-effects)))
```

Reply:
And if you want to have a lot of fun, this talk about a condition system implemented in Clojure is too interesting https://youtu.be/zp0OEDcAro0?si=ssDADWNnSYzgam8a. Be aware of the rabbit hole!